### PR TITLE
Added option to emit final annotations

### PR DESCRIPTION
**Pull request by azidar**
Added option to emit final annotations to a file.

### CODE REVIEW

File: src/main/scala/firrtl/ExecutionOptionsManager.scala [hunk 1]
```diff
@@ -298,6 +299,15 @@ trait HasFirrtlOptions {
       "use this to force appending annotation file to annotations being passed in through optionsManager"
     }
 
+  parser.opt[String]("output-annotation-file")
+    .abbr("foaf")
```
Comment (edwardcwang) [inline src/main/scala/firrtl/ExecutionOptionsManager.scala]: What does foaf mean?

File: src/main/scala/firrtl/ExecutionOptionsManager.scala [hunk 2]
```diff
@@ -283,8 +284,8 @@ trait HasFirrtlOptions {
 
   parser.opt[String]("annotation-file")
     .abbr("faf")
-    .valueName ("<output>").
-    foreach { x =>
+    .valueName ("<input-anno-file>>")
```
Comment (edwardcwang) [inline src/main/scala/firrtl/ExecutionOptionsManager.scala]: Is this intended to have an extra right bracket?

**Comment by azidar** [comment]
Re freechipsproject/chisel3#566, i think that was for dumping annotations before FIRRTL compilation, while this dumps annotations after FIRRTL compilation. I'll make the small changes and merge.

File: src/main/scala/firrtl/ExecutionOptionsManager.scala (hunk 1 above)
Comment (azidar) [reply src/main/scala/firrtl/ExecutionOptionsManager.scala]: foaf = firrtl output annotation file, its like the previous style for shortened commandline flags

File: src/main/scala/firrtl/ExecutionOptionsManager.scala (hunk 2 above)
Comment (azidar) [reply src/main/scala/firrtl/ExecutionOptionsManager.scala]: nope, i'll fix it.

**Comment by edwardcwang** [comment]
Sounds good; LGTM.

**Pull request merged**